kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine
+
+import scala.collection.JavaConverters._
+
+import org.apache.hadoop.yarn.client.api.YarnClient
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.ApplicationOperation._
+import org.apache.kyuubi.util.KyuubiHadoopUtils
+
+class YarnApplicationOperation extends ApplicationOperation with Logging {
+
+  @volatile private var yarnClient: YarnClient = _
```

Comment:
It seems that the `YarnApplicationOperation` is singleton, should we use a global yarnClient?


Is it better to use temporary yarnClient for per request?

Reply:
I guess you are suggesting per thread not per request?

Reply:
thread shared yarn client sounds good

Reply:
We can add ThreadedYarnApplicationOperation later. For most use case the current implementation is enough